### DISCUSSION TITLE
Standardize and Automate Our Bug Report Template

### DISCUSSION THREAD

**Discussion by SirTypesALot** [organicmaps/organicmaps#2225] [Ideas]
I'm going through some of the bug reports made in Issues, and I can't help but notice that we can significantly improve the bug reporting workflow and efficiency if we made sure that the initial report would already contain all the necessary debugging information to start with.

I see that the repo already has a bug report section in the contribution file, however, it's a bit hidden with all of the other files. Making sure that all new bug reports starts from the bug report template form would minimize this issue.

We can [configure](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository) that all new bug reports automatically starts with a template bug report form

This is how I would set up the bug report template:

- Title
- Environment (Device, OS, App version etc.)
- Steps to reproduce
- Actual Result
- Expected Result
- Visual Proof (screenshots etc.)
- Severity/Priority (minor, medium, high/critical)

**Comment by SirTypesALot** [comment]
The same could be applied for feature requests

**Reply by biodranik** [reply]
The idea is good, but in practice, it is not possible to fit everyone into a template. Also, many issues are cross-platform. We have many reports and issues created by non-technical people, and even in different languages. It can be more efficient to post-edit a non-perfect issue than to miss it completely because someone failed to understand and follow the template.

What can be helpful though is a hint to put as much information about an issue as possible.

**Reply by SirTypesALot** [reply]
Indeed, what we can do then is to make the template of a bug report to simply contain a 'warning' or a/few sentences asking the reporter to contain as much information possible. Not necessarily in a specific format, but as long as the initial post contains all the data then it's good.

My point is that we should make it more obvious for the bug reporters to report all the necessary information in one go.

**Reply by biodranik** [reply]
Can you please propose a PR with your suggestions? Just try to make it simple for non-geeks.

**Reply by SirTypesALot** [reply]
Sure, I can do a PR with my suggestions, but ultimately we would still have to add the automatic bug report 'template' with the follow description so that everyone would see it.

We can start with the basic information:
- A description of the bug (with screenshots if possible)
- The user's environment: device model, OS version and app version (if it's a cross platform bug, then list all of the devices that are confirmed to be affected)

And then for the more technical contributors:
- Step by step description on how we can reproduce the bug
- Expected and actual results
- Severity
